game-core/src/test/java/games/strategy/triplea/delegate/battle/steps/fire/aa/FiringGroupFilterAaTest.java

```diff
@@ -0,0 +1,351 @@
+package games.strategy.triplea.delegate.battle.steps.fire.aa;
+
+import static games.strategy.triplea.Constants.UNIT_ATTACHMENT_NAME;
+import static games.strategy.triplea.delegate.battle.BattleState.Side.DEFENSE;
+import static games.strategy.triplea.delegate.battle.BattleState.Side.OFFENSE;
+import static games.strategy.triplea.delegate.battle.FakeBattleState.givenBattleStateBuilder;
+import static games.strategy.triplea.delegate.battle.steps.BattleStepsTest.givenAnyUnit;
+import static games.strategy.triplea.delegate.battle.steps.BattleStepsTest.givenUnitIsCombatAa;
+import static games.strategy.triplea.delegate.battle.steps.BattleStepsTest.givenUnitIsInfrastructure;
+import static games.strategy.triplea.delegate.battle.steps.MockGameData.givenGameData;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.contains;
+import static org.hamcrest.Matchers.hasSize;
+import static org.hamcrest.Matchers.is;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import games.strategy.engine.data.GameData;
+import games.strategy.engine.data.GamePlayer;
+import games.strategy.engine.data.MutableProperty;
+import games.strategy.engine.data.Unit;
+import games.strategy.triplea.Constants;
+import games.strategy.triplea.attachments.TechAbilityAttachment;
+import games.strategy.triplea.attachments.TechAttachment;
+import games.strategy.triplea.attachments.UnitAttachment;
+import games.strategy.triplea.delegate.TechAdvance;
+import games.strategy.triplea.delegate.battle.steps.fire.FiringGroup;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@ExtendWith(MockitoExtension.class)
+class FiringGroupFilterAaTest {
+
+  @Mock GamePlayer attacker;
+  @Mock GamePlayer defender;
+
+  @Test
+  void basicFiringGroup() {
+    final Unit targetUnit = givenAnyUnit();
+    final Unit fireUnit = givenUnitIsCombatAa(Set.of(targetUnit.getType()), attacker, OFFENSE);
+    when(fireUnit.getOwner()).thenReturn(attacker);
+
+    final List<FiringGroup> firingGroups =
+        FiringGroupFilterAa.of(OFFENSE)
+            .apply(
+                givenBattleStateBuilder()
+                    .gameData(givenGameData().withWarRelationship(defender, attacker, true).build())
+                    .attacker(attacker)
+                    .defender(defender)
+                    .attackingUnits(List.of(fireUnit))
+                    .defendingUnits(List.of(targetUnit))
+                    .build());
+
+    assertThat(firingGroups, hasSize(1));
```

Comment:
Why should the firing group have a size of one?

"Have pity on the reader"

I forget who wrote that, I think it was Fowler perhaps, but have pity on the reader of code and make the things that are obvious, *really* obvious. A reader does not have context and is thinking about several concerns. An assertion comment string here could go a long way to explain why this assertion should be a true statement. EG:
```
assertThat("combat AA can taget any unit, given one opposing unit, expect that unit to be in the firing group", firingGroups, hasSize(1))
```

Reading this test, that is my best guess why this assertion is valid. Beyond that, I'd have to start debugging and/or reverse engineering the code. If there were any changes that caused this test to break, that is what a developer would have to do, to fix this one assertion, of this one test. I emphasize this for the importance of making it very easy for maintainers to read tests and understand why the test is valid. A mixture of `@DisplayText` and assertion comments should be favored to help explain this. When there is one assertion, a `@DisplayText` is sufficient, when there are multiple assertions then generally assertion comments should start coming into play. (For the record this is a convention that should be followed, to make tests easy to understand for maintainers: 
- https://github.com/triplea-game/triplea/blob/master/docs/development/code-conventions/java-test-code-conventions.md#favor-adding-displayname-to-each-test-case-5525
- https://github.com/triplea-game/triplea/blob/master/docs/development/code-conventions/java-test-code-conventions.md#use-assertthat-string-parameter-to-describe-why-we-assert-a-condition-5527)